src/core/qgsapplication.h

```diff
@@ -1024,6 +1032,13 @@ class CORE_EXPORT QgsApplication : public QApplication
      */
     void requestForTranslatableObjects( QgsTranslationContext *translationContext );
 
+
+    /**
+     * Emitted when project locale has been changed.
```

Comment:
Forgot to ask, can you add `\since QGIS 3.24`

Reply:
Yes, I will add the since clause.